src/main/java/com/redislabs/redisgraph/RedisGraph.java

```diff
@@ -0,0 +1,192 @@
+        this.client.close();
+    }
+
+
+    /**
+     * Execute a Cypher query with arguments
+     *
+     * @param graphId a graph to perform the query on
+     * @param query Cypher query
+     * @param args
+     * @return a result set
+     */
+    public ResultSet query(String graphId, String query, Object ...args) {
+        if(args.length > 0) {
+            for(int i=0; i<args.length; ++i) {
+                if(args[i] instanceof String) {
+                    args[i] = "\'" + ESCAPE_CHYPER.translate((String)args[i]) + "\'";
+                }
+            }
+            query = String.format(query, args);
+        }
+        graphCaches.putIfAbsent(graphId, new GraphCache(graphId, this));
+
+        try (Jedis conn = getConnection()) {
+            List<Object> rawResponsw = sendCompactCommand(conn, Command.QUERY, graphId, query).getObjectMultiBulkReply();
+            conn.close();
+            return new ResultSetImpl(rawResponsw, graphCaches.get(graphId));
+        }
+    }
+
+    /**
+     * Invokes stored procedures without arguments
+     * @param graphId a graph to perform the query on
+     * @param procedure procedure name to invoke
+     * @return result set with the procedure data
+     */
+    public ResultSet callProcedure(String graphId, String procedure  ){
+        return callProcedure(graphId, procedure, new ArrayList<>(), new HashMap<>());
+    }
+
+
+    /**
+     * Invokes stored procedure with arguments
+     * @param graphId a graph to perform the query on
+     * @param procedure procedure name to invoke
+     * @param args procedure arguments
+     * @return result set with the procedure data
+     */
+    public ResultSet callProcedure(String graphId, String procedure, List<String> args  ){
+        return callProcedure(graphId, procedure, args, new HashMap<>());
+    }
+
+
+    /**
+     * Deletes the entire graph
+     *
+     * @return delete running time statistics
+     */
+    public String deleteGraph(String graphId) {
+        //clear local state
+        graphCaches.remove(graphId);
+        try (Jedis conn = getConnection()) {
+            return sendCommand(conn, Command.DELETE, graphId).getBulkReply();
+        }
+
+    }
+
+
+    /**
+     * Sends command - will be replaced with sendCompactCommand once graph.delete support --compact flag
+     * @param conn - connection
+     * @param provider - command type
+     * @param args - command arguments
+     * @return
+     */
+    private BinaryClient sendCommand(Jedis conn, ProtocolCommand provider, String ...args) {
+        BinaryClient binaryClient = conn.getClient();
+        binaryClient.sendCommand(provider, args);
+        return binaryClient;
+    }
+
+
+    /**
+     * Sends the command with --COMPACT flag
+     * @param conn - connection
+     * @param provider - command type
+     * @param args - command arguments
+     * @return
+     */
+    private BinaryClient sendCompactCommand(Jedis conn, ProtocolCommand provider, String ...args) {
```

Comment:
Feels like some code duplication, isn't cleaner this way?
```
        String[] comArgs = new String[args.length +1];
        System.arraycopy(comArgs, 0 , t, 0, args.length);
        comArgs[args.length]="--COMPACT";
       sendCommand(conn, provider, t)
```

Reply:
done